Form for creating and editing collections

awead opened the pull request:
Implements a user interface for creating and editing collections. It duplicates the same workflow and design as the work version's interface. Because of the complexity of the interface, and in order to avoid a considerable amount of code duplication, there is a large amount of refactoring, namely re-naming the original work form interface module and redesigning the urls to accommodate both work versions and collections.

The original collection interface is also removed. It had been implemented prior to design work from West Arete, and was never intended for production release.

Fixes #742
Fixes #744

cam156 commented on spec/features/generic_work/upload_and_delete_spec.rb:
```diff
@@ -49,14 +49,11 @@
 
-        # Check for additional fields
-        expect(page).to have_no_css("#generic_work_contributor")
-        click_link("Additional fields")
```
@awead this used to check that the additional fields opened up when clicked.  Why would we want to remove that check?

awead replied:
The animation wasn't working in the test so I took a different tack that we've using in other tests and don't test the animation. Instead, I'm testing that they're hidden.

cam156 commented on spec/support/shared/forms.rb:
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
```
Cool!

awead committed:
Collection form for managing collections

Duplicates the same user interface and workflow that's used to create
and edit works. Included in this is a MASSIVE refactoring of all the
controller and view code that was in the original work form. In order to
reduce the duplication of controller and view code, the previous
WorkForm module is renamed to a Form module, and incorporates collection
controllers as well.

Controllers for contributors are identical for both work versions and
collections, while details are more similar and the view layer.

Routing is now slightly different, eschewing the /dashboard/work-form
path in favor of /dashboard/form/work_version or
/dashboard/form/collection.

Duplicates the work form controllers and views in order to build a form
for creating new collections.

awead committed:
Add indexing source to collections

Mirroring work versions, the indexing option allows collections to be
indexed either asynchronously (the default) or synchronously by
specifying a different option.

awead committed:
Add position column to CollectionCreation

This duplicates 3c338cbdc24f508c6e3abcb3e79d4c14d5c8df2c and adds the
ability to explicit order the creators that are linked to a collection
in the same way we order creators linked to a work version.

awead committed:
Add proxy-owned works to the dashboard

Dashboard search now includes works that have been proxy-deposited by a
user. While there isn't a UI method (yet) for depositing new works on
behalf of someone, or otherwise identifying a proxy depositor, this
covers existing migrated works that have this field in use.

awead committed:
Removing old collection code

This was the previous incarnation of collection management, which was
never officially released.
